.github/workflows/ci-notebooks.yml

```diff
@@ -0,0 +1,35 @@
+name: ci-notebooks
+on:
+  pull_request:
+    paths:
+      - modin/**
+      - examples/tutorial/tutorial_notebooks/introduction/**
+jobs:
+  test-notebooks:
+    name: test notebooks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+      - uses: actions/setup-python@v2
+        with:
```

Comment:
I would probably prefer to use strategy -> matrix to facilitate further extensions.

Reply:
Does that make sense?

Reply:
That depends on the way user should set the modin engine  - using environment variable outside the notebook or using `modin.config`. I prefer the last option - for me it seems more convenient from the user's point of view, and, in this case new engines should be added as `pytest` parameters or test modification in the `test_notebooks.py`.

Reply:
That depends on us, users are not supposed to test notebooks with our tests. I pondered the use of strategy -> matrix is not a good option because we will have a separate env file for each engine. So I think separate jobs would work. What do you think?

Reply:
Yes, sure, the way user should specify engine setting is up to us, but in this point we have two options to make it:
1) Write in the notebook something like 'if you want to try out this notebook with another engine, you should set environment variable MODIN_ENGINE and restart notebook'. In this case we really should use strategy -> matrix with tested engines.
2) Add commented code in the notebooks which activates alternative engine as it is done [here](https://github.com/modin-project/modin/pull/4139/files#diff-0912c12d3ebc727b8ab46b5b4ad258903c7d0b76d0dbc76d41f49086f9ff676bR75-R76). In this case we need to test the engine switching technique we had proposed to the user, so this should be done by code replacement (uncommenting of the code with engine setting) in the `test_notebooks.py`.

The second option seems for me more preferable and as for now it doesn't require strategy -> matrix usage in the CI file. So what is purpose of jobs separation you proposed (currently new tests are finish pretty fast)? And what is the purpose of separated requirements file for different engines? Currently `modin[all]` stated in the requirements file, so all engines should be installed.

Reply:
I prefer each scope of notebooks would be tied to a specific `StorageFormat` and a specific `Engine`.

> And what is the purpose of separated requirements file for different engines? Currently modin[all] stated in the requirements file, so all engines should be installed.

modin[all] will install deps for all engines except OmniSci so I think we should have a separate file for each component.